src/libs/histogram.c

```diff
@@ -458,6 +505,52 @@ static void _lib_histogram_vectorscope_bkgd(dt_lib_histogram_t *d, const dt_iop_
   d->hue_ring_colorspace = d->vectorscope_type;
 }
 
+static  void _get_chromaticity(const dt_aligned_pixel_t RGB, dt_aligned_pixel_t chromaticity,
+                               const dt_lib_histogram_vectorscope_type_t vs_type,
+                               const dt_iop_order_iccprofile_info_t *vs_prof,
+                               const float *rgb2ryb_ypp)
+{
+  dt_aligned_pixel_t XYZ_D50;
+  if(vs_type != DT_LIB_HISTOGRAM_VECTORSCOPE_RYB)
```

Comment:
let's simplify the code here with a swicth:

```
switch(vs_type)
{
   case DT_LIB_HISTOGRAM_VECTORSCOPE_RYB:
   ...
```
And yes I'll duplicate the call dt_ioppr_rgb_matrix_to_xyz() in each alternatives maybe even the declaration of XYZ_D50 which is not used for the RYB case.